sympy/polys/polyroots.py

```diff
@@ -343,6 +341,10 @@ def _ans(y):
                         ans.append((s*w - t*root)/2 - aon4)
                 return ans
 
+            # whether a Piecewise is returned or not
+            # depends on knowing p, so try to simplify
+            p = p.simplify()  # XXX only if number?
```

Comment:
Here `p` comes from `p = -e**2/12 - g` above. The reason this simplify is added is because if `e` is an `Add` like `1 + sqrt(2)` then we need to expand the power to see whether it is zero. I think it is better to use `_mexpand` here since it is clear why that might be needed in some cases.

There might be some cases that get left unresolved where `simplify` could handle them but in those cases we return a Piecewise which is still correct. The caller can use simplify if they want. Ideally `refine` would be better at handling this sort of thing or there should be a `simplify_piecewise` function whose job is precisely to simplify the conditional parts of a Piecewise.